### ISSUE TITLE
Module installation, disable not yet installed module on exception

### ISSUE THREAD

**Issue by globox91** [PrestaShop/PrestaShop#16788]
**Describe the bug**
`In "importModuleAction" (\PrestaShopBundle\Controller\Admin\Improve\ModuleController::importModuleAction)`
there is a line that disables module on any thrown Exception.
To disable a module, I need an access for a "configure" action on that module 
`(\PrestaShop\PrestaShop\Adapter\Module\AdminModuleDataProvider::isAllowedAccess)`
To get info about my permissions for a module, Presta tries to get a module Id first, using its name
`(\PrestaShop\PrestaShop\Adapter\Module\ModuleDataProvider::can)`

But there can be cases when exceptions on install will be thrown before module name is even inserted into "ps_module" table. So there is no module Id yet. Then, I'm automatically considered to miss required access to "disabling" that module. And installation just throws "You are not allowed to disable the module %module%" error, which is, imho, absolutely wrong :)

**To Reproduce**
1. Try to install any module by uploading a zip
[v4.6.4-chronopost.zip](https://github.com/PrestaShop/PrestaShop/files/3960471/v4.6.4-chronopost.zip)
2. Throw Exception before install process inserts module name into "ps_module" table
3. See wrong error message => message displayed is wrong
4. you can check the good message by copying the folder module unzipped into the path Project_Folder/modules/ & trying to install it from the BO => Module Catalog page => OK ( good message is displayed)

**Additional information**
PrestaShop version: 1.7.6.1
PHP version: 5.6

**Comment by khouloudbelguith** [comment]
Hi @globox91,

What is the exact name of this module?
Do you have an exception during the installation of this module?
Thanks to provide me a screenshot.

Thanks!

**Comment by globox91** [comment]
It's "BNP Parisbas - Mercanet", but it's really irrelevant, cause, as I mentioned above, any module can throw an Exception during any moment of the install process :)

You need a screenshot of an Exception itself? o_O

**Comment by khouloudbelguith** [comment]
@globox91, so, during the install of your module which fails with an exception, you have a wrong alert
"You are not allowed to disable the module"? this error is wrong?
Have you tried to contact the module author about this issue?

Thanks!

**Comment by globox91** [comment]
Yep, instead of an Exception message, I'm failing to disable the module and get message about that :)

Nope, I haven't, cause it's not an issue with the module - it still tries to display the correct Exception message. But Presta interrupts this with "disable" operation, which cannot be executed on a non-installed module ^^

**Comment by khouloudbelguith** [comment]
@globox91, Could you please provide me the module to test it.
This is a public space.
You can provide me the module by email.
My address mail: khouloud.belguith@prestashop.com
Let's be safe!

Thanks!

**Comment by globox91** [comment]
Done :)

**Comment by globox91** [comment]
But I can easily imagine it will be installed without errors cause Exception is being thrown because of my shop config error :)

But as I said, I cannot see that Exception message due to Presta logic

**Comment by globox91** [comment]
Step 1 of "To Reproduce" should be "Try to install any module"

I'll fix that

**Comment by khouloudbelguith** [comment]
@globox91, thanks!
I will check & feedback.

**Comment by khouloudbelguith** [comment]
@globox91, it is well installed
https://drive.google.com/file/d/1roXQ81-7xfR2shPA7pHiRgIvGCTYia8R/view

Thanks!

**Comment by globox91** [comment]
> But I can easily imagine it will be installed without errors cause Exception is being thrown because of my shop config error :)
> 
> But as I said, I cannot see that Exception message due to Presta logic

All you need to check this bug report is to throw any Exception in any modules' "install" method :)

For example, you can try on this module too.
You will not see the message of your Exception

**Comment by khouloudbelguith** [comment]
@globox91, I tried with this module, is not compatible with PS1.7
[v4.6.4-chronopost.zip](https://github.com/PrestaShop/PrestaShop/files/3959820/v4.6.4-chronopost.zip)
I have an exception during the install & it is not wrong
https://drive.google.com/file/d/1TEx0gMHOzOZHWTZ1UCXW7tvfvHJg7_Z9/view

Thanks to check & feedback.

**Comment by globox91** [comment]
And I managed to successfully install it :D

![изображение](https://user-images.githubusercontent.com/56883311/70791180-a698d800-1da7-11ea-8816-fdf727647720.png)

Okey, let's stick to this module.
I'll update the bug description: you will receive it if the Exception is thrown before "parent::install()" call.
In "mercanet" module it's called right in the end of the "install" method.
And in "chronopost" it's called in "preInstall" method.


[v4.6.4-chronopost.zip](https://github.com/PrestaShop/PrestaShop/files/3960211/v4.6.4-chronopost.zip)

I added two Exceptions in "chronopost" module file. Both are in "preInstall" method.
One sits before "parent::install()" call, and other is placed after that call.
Second is commented out.

Now during the installation you should see the "disabled" error text, not the text of my first exception.

![изображение](https://user-images.githubusercontent.com/56883311/70791909-24111800-1da9-11ea-9ce1-0bcef198d048.png)

And if you comment out the first exception and uncomment the second, you will see the correct text of that second exception :)

![изображение](https://user-images.githubusercontent.com/56883311/70791924-2d9a8000-1da9-11ea-8252-1c206f75b8e5.png)

I hope now we are on the same wave ^^

**Comment by khouloudbelguith** [comment]
@globox91, thanks for these clarifications.
I manage to reproduce the issue with PS1.7.6.2 & PS1.7.5.2.
![image](https://user-images.githubusercontent.com/16067358/70795755-d85a7080-1da0-11ea-9585-74b5f8907db2.png)
To reproduce the issue, you need to upload the module from the Back office
If we copy the folder module into the path Project_Folder/modules/ & we try to install it from the BO => Module Catalog page => OK
![image](https://user-images.githubusercontent.com/16067358/70796121-c7f6c580-1da1-11ea-8c92-fc23a4987b55.png)
I’ll add this to the debug roadmap so that it’s fixed. If you have already fixed it on your end or if you think you can do it, please do send us a pull request!
Thanks!

**Comment by ernest98k** [comment]
Did you fix it? 
I have the same issue..

**Comment by khouloudbelguith** [comment]
@globox91,

Sorry not yet.
Thanks!

**Comment by damcoll** [comment]
Hello,
I have the same issue.
Did you find a solution ?
I can be a solution.
In ModuleController.php ligne 614 
`$moduleManager->disable($moduleName);`

i have change by 
`try {
$moduleManager->disable($moduleName);
} catch (Exception $disableError){
}`
I know it's not very clean but now I have the errors of my module.
This allowed me to correct them and to be able to install my module

sorry for my English, hoping to have been useful to you

**Comment by caglarutkuguler** [comment]
Same error with Prestashop 1.7.6.7

No soluiton is provided, yet.
